src/cat.jl

```diff
@@ -0,0 +1,24 @@
+function equalvalued(X::NTuple)
+    n = length(X)
+    allequal = true
+    i = 2
+    while allequal && i <= n
+        allequal = X[i] == X[i-1]
+        i += 1
+    end #while
+    return allequal
+end #equalvalued
+
+sizes{T<:AxisArray}(As::T...) = tuple(zip(map(size, As)...)...)
+matchingdims{N,T<:AxisArray}(As::NTuple{N,T}) = all(equalvalued, sizes(As...))
+matchingdimsexcept{N,T<:AxisArray}(As::NTuple{N,T}, n::Int) = all(equalvalued, sizes(As[[1:n-1; n+1:end]]...))
+
+Base.cat{T<:AxisArray}(n::Int, As::T...) = if n <= ndims(As[1])
+    matchingdimsexcept(As, n) || error("All non-concatenated axes must be identically-valued")
+    newaxis = Axis{axisnames(As[1])[n]}(vcat(map(A -> A.axes[n].val, As)...))
+    checkaxis(newaxis)
```

Comment:
Ah, so just to confirm — if you concatenate along a dimension where each array has a Range axis, this just calls `vcat`  on those ranges, returning a dense vector, yes?  That's a great choice — it allows further optimizations down the road but doesn't lock us into any particular behavior at this time.